File: Rentify.Services/Database/Payment.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Rentify.Services.Database
{
    public class Payment
    {
        [Key]
        public int Id { get; set; }

        [ForeignKey(nameof(UserId))]
        public int UserId { get; set; }
        public User? User { get; set; }

        [ForeignKey(nameof(PropertyId))]
        public int PropertyId { get; set; }
        public Property? Property { get; set; }
        public double Price { get; set; }
        public bool IsPayed { get; set; }
        public DateTime? DateToPay { get; set; }
        public DateTime? WarningDateToPay { get; set; }
    }
}

File: UI/rentify_mobile/lib/models/payment.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:rentify_mobile/models/property.dart';
import 'package:rentify_mobile/models/user.dart';

part 'payment.g.dart';

@JsonSerializable()
class Payment {
  final int id;
  final int userId;
  User? user;
  final int propertyId;
  Property? property;
  final String name;
  final String comment;
  final double price;
  final int monthNumber;
  final int yearNumber;
  final bool isPayed;
  DateTime? dateToPay;
  DateTime? warningDateToPay;  

  Payment({
    required this.id,
    required this.userId,
    this.user,
    required this.propertyId,
    this.property,
    required this.name,
    required this.comment,
    required this.price,
    required this.isPayed,
    required this.monthNumber,
    required this.yearNumber,
    this.dateToPay,
    this.warningDateToPay,
  });

  factory Payment.fromJson(Map<String, dynamic> json) =>
      _$PaymentFromJson(json);

  Map<String, dynamic> toJson() => _$PaymentToJson(this);
}
